server/api/receipt_controller.py

```diff
@@ -34,23 +34,50 @@ def create():
     return jsonify({"Success": "New receipt created"})
 
 
-@receipt_controller.route('/', defaults={'year': None, 'month': None},
-                          methods=['GET'])
+@receipt_controller.route('/', methods=['GET'])
+@receipt_controller.route('/<int:year>', methods=['GET'])
 @receipt_controller.route('/<int:year>/<int:month>', methods=['GET'])
-def get_all_receipts(year, month):
+@receipt_controller.route('/<int:year>/<int:month>/<int:date>', methods=['GET'])
+def get_all_receipts(year=None, month=None, date=None, weekly=False):
+    if request.args.get('weekly') is not None:
+        weekly = int(request.args.get('weekly'))
     total_amount = 0
     receipts = []
 
-    if year is None and month is None:
-        receipts = Receipt.query.all()
+    if year is None and month is None and date is None:
+        receipts = Receipt.query.order_by(Receipt.receipt_date.desc()).all()
         # Receipt.query.all() returns empty list if there are no records
         if receipts == []:
             return receipts
-    elif year is not None and month is not None:
+    elif year is not None and month is None and date is None:
+        start_date = datetime.date(year, 1, 1)
+        end_date = datetime.date(year, 12, 31)
+
+        receipts = (Receipt.query
+                    .filter(Receipt.receipt_date <= end_date)
+                    .filter(Receipt.receipt_date >= start_date)
+                    .order_by(Receipt.receipt_date.desc()))
+    elif year is not None and month is not None and date is None:
         num_days = calendar.monthrange(year, month)[1]
         start_date = datetime.date(year, month, 1)
         end_date = datetime.date(year, month, num_days)
 
+        receipts = (Receipt.query
+                    .filter(Receipt.receipt_date <= end_date)
```

Comment:
a lot of this same filter 

```
receipts = (Receipt.query
                    .filter(Receipt.receipt_date <= end_date)
                    .filter(Receipt.receipt_date >= start_date)
                    .order_by(Receipt.receipt_date.desc()))
```

is the same. to reduce the # of lines, you can perhaps look into creating a helper func

Reply:
I realized that pretty much all of the receipt queries could be formatted into the same filter so I made some changes to the code in some of the IF blocks so all of them can use the same filter.